Compilation error on Clang 12.0.0

**Describe the bug**
Compilation error on Clang 12.0.0:
```
Assimp/include/assimp/TinyFormatter.h:122:20: error: invalid operands to binary expression ('Assimp::Formatter::basic_formatter<char, std::char_traits<char>, std::allocator<char>>::stringstream' (aka 'basic_ostringstream<char, std::char_traits<char>, std::allocator<char>>') and 'const DeadlyImportError')
```
The offending file which includes the formatter is:
```
Assimp/code/PostProcessing/DropFaceNormalsProcess.cpp
```

**To Reproduce**
Steps to reproduce the behavior:
1. Compile on Windows with Clang 12.0.0.

**Expected behavior**
A clear and concise description of what you expected to happen.
1. Compilation should have succeeded.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: Windows 10 build 20H2

It could just be an incompatibility with Clang 12 and the MSVC standard library, but I don't think that is the case.
It seems like it's trying to log a `DeadlyImportError` into an output stream.

Here's the entire error:
```
In file included from ../../External/Assimp/code/PostProcessing/DropFaceNormalsProcess.cpp:52:
In file included from ../../External/Assimp/include\assimp/DefaultLogger.hpp:49:
In file included from ../../External/Assimp/include\assimp/Logger.hpp:50:
../../External/Assimp/include\assimp/TinyFormatter.h:122:20: error: invalid operands to binary expression ('Assimp::Formatter::basic_formatter<char, std::char_traits<char>, std::allocator<char>>::stringstream' (aka 'basic_ostringstream<char, std::char_traits<char>, std::allocator<char>>') and 'const DeadlyImportError')
        underlying << s;
        ~~~~~~~~~~ ^  ~
../../External/Assimp/include\assimp/Exceptional.h:65:41: note: in instantiation of function template specialization 'Assimp::Formatter::basic_formatter<char, std::char_traits<char>, std::allocator<char>>::operator<<<DeadlyImportError>' requested here
            DeadlyErrorBase(std::move(f << std::forward<U>(u)), std::forward<T>(args)...) {}
                                        ^
../../External/Assimp/include\assimp/Exceptional.h:77:13: note: in instantiation of function template specialization 'DeadlyErrorBase::DeadlyErrorBase<, DeadlyImportError &>' requested here
            DeadlyErrorBase(Assimp::Formatter::format(), std::forward<T>(args)...) {}
            ^
../../External/Assimp/code/PostProcessing/DropFaceNormalsProcess.cpp:83:15: note: in instantiation of function template specialization 'DeadlyImportError::DeadlyImportError<DeadlyImportError &>' requested here
        throw DeadlyImportError("Post-processing order mismatch: expecting pseudo-indexed (\"verbose\") vertices here");
              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:653:32: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'const char *' for 2nd argument
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr, const char* _Val) { // insert NTBS
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:697:32: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'char' for 2nd argument
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr, char _Ch) { // insert a character
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:733:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'const char *' for 2nd argument
basic_ostream<char, _Traits>& operator<<(basic_ostream<char, _Traits>& _Ostr,
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:778:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'char' for 2nd argument
basic_ostream<char, _Traits>& operator<<(
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:895:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'const signed char *' for 2nd argument
basic_ostream<char, _Traits>& operator<<(
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:901:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'signed char' for 2nd argument
basic_ostream<char, _Traits>& operator<<(basic_ostream<char, _Traits>& _Ostr, signed char _Ch) { // insert a signed char
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:906:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'const unsigned char *' for 2nd argument
basic_ostream<char, _Traits>& operator<<(
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:912:31: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'unsigned char' for 2nd argument
basic_ostream<char, _Traits>& operator<<(
                              ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:998:32: note: candidate function template not viable: no known conversion from 'const DeadlyImportError' to 'const std::error_code' for 2nd argument
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr,
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:859:32: note: candidate template ignored: deduced conflicting types for parameter '_Elem' ('char' vs. 'DeadlyImportError')
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr, _Elem _Ch) { // insert a character
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\xstring:4724:32: note: candidate template ignored: could not match 'basic_string<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'const DeadlyImportError'
basic_ostream<_Elem, _Traits>& operator<<(
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\memory:2077:32: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-2>' against 'const DeadlyImportError'
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Out, const shared_ptr<_Ty>& _Px) {
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\memory:3718:32: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-2, type-parameter-0-3>' against 'const DeadlyImportError'
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Out, const unique_ptr<_Yty, _Dx>& _Px) {
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:816:32: note: candidate template ignored: could not match 'const _Elem *' against 'DeadlyImportError'
basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr, const _Elem* _Val) { // insert NTCS
                               ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:970:9: note: candidate template ignored: substitution failure [with _Ostr = std::basic_ostringstream<char, std::char_traits<char>, std::allocator<char>> &, _Ty = DeadlyImportError]: 'type name' declared as a pointer to a reference of type 'std::basic_ostringstream<char, std::char_traits<char>, std::allocator<char>> &'
_Ostr&& operator<<(_Ostr&& _Os, const _Ty& _Val) { // insert to rvalue stream
        ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:183:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'std::basic_ostream<char, char_traits<char>> &(*)(std::basic_ostream<char, char_traits<char>> &) __attribute__((cdecl))' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:188:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'std::basic_ostream<char, std::char_traits<char>>::_Myios &(*)(std::basic_ostream<char, std::char_traits<char>>::_Myios &) __attribute__((cdecl))' (aka 'basic_ios<char, std::char_traits<char>> &(*)(basic_ios<char, std::char_traits<char>> &)') for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(_Myios&(__cdecl* _Pfn)(_Myios&) ) { // call basic_ios manipulator
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:193:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'std::ios_base &(*)(std::ios_base &) __attribute__((cdecl))' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(ios_base&(__cdecl* _Pfn)(ios_base&) ) { // call ios_base manipulator
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:198:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'bool' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(bool _Val) { // insert a boolean
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:216:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'short' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(short _Val) { // insert a short
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:250:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'unsigned short' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(unsigned short _Val) { // insert an unsigned short
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:269:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'int' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(int _Val) { // insert an int
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:295:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'unsigned int' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(unsigned int _Val) { // insert an unsigned int
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:314:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'long' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(long _Val) { // insert a long
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:332:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'unsigned long' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(unsigned long _Val) { // insert an unsigned long
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:350:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'long long' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(long long _Val) { // insert a long long
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:368:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'unsigned long long' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(unsigned long long _Val) { // insert an unsigned long long
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:386:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'float' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(float _Val) { // insert a float
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:404:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'double' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(double _Val) { // insert a double
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:422:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'long double' for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(long double _Val) { // insert a long double
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:440:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'const void *' for 1st argument; take the address of the argument with &
    basic_ostream& __CLR_OR_THIS_CALL operator<<(const void* _Val) { // insert a void pointer
                                      ^
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.28.29812\include\ostream:465:39: note: candidate function not viable: no known conversion from 'const DeadlyImportError' to 'std::basic_ostream<char, std::char_traits<char>>::_Mysb *' (aka 'basic_streambuf<char, std::char_traits<char>> *') for 1st argument
    basic_ostream& __CLR_OR_THIS_CALL operator<<(_Mysb* _Strbuf) { // insert until end-of-file from a stream buffer
                                      ^
1 error generated.
```

--- a/include/assimp/Exceptional.h
+++ b/include/assimp/Exceptional.h
@@ -98,10 +98,6 @@ public:
             DeadlyErrorBase(Assimp::Formatter::format(), std::forward<T>(args)...) {
         // empty
     }
-
-#if defined(_MSC_VER) && defined(__clang__)
-    DeadlyImportError(DeadlyImportError& other) = delete;
-#endif
 };
 
 // ---------------------------------------------------------------------------
@@ -114,10 +110,6 @@ public:
     template<typename... T>
     explicit DeadlyExportError(T&&... args) :
             DeadlyErrorBase(Assimp::Formatter::format(), std::forward<T>(args)...) {}
-
-#if defined(_MSC_VER) && defined(__clang__)
-    DeadlyExportError(DeadlyExportError& other) = delete;
-#endif
 };
 
 #ifdef _MSC_VER

--- a/include/assimp/TinyFormatter.h
+++ b/include/assimp/TinyFormatter.h
@@ -119,15 +119,27 @@ public:
      * work for const references, so many function prototypes will
      * include const basic_formatter<T>& s but might still want to
      * modify the formatted string without the need for a full copy.*/
-    template <typename TToken>
-    const basic_formatter& operator << (const TToken& s) const {
+    template <typename TToken, typename std::enable_if<!std::is_base_of<std::exception, TToken>::value>::type * = nullptr>
+    const basic_formatter &operator<<(const TToken &s) const {
         underlying << s;
         return *this;
     }
 
-    template <typename TToken>
-    basic_formatter& operator << (const TToken& s) {
+    template <typename TToken, typename std::enable_if<std::is_base_of<std::exception, TToken>::value>::type * = nullptr>
+    const basic_formatter &operator<<(const TToken &s) const {
+        underlying << s.what();
+        return *this;
+    }
+
+    template <typename TToken, typename std::enable_if<!std::is_base_of<std::exception, TToken>::value>::type * = nullptr>
+    basic_formatter &operator<<(const TToken &s) {
         underlying << s;
+        return *this;
+    }
+
+    template <typename TToken, typename std::enable_if<std::is_base_of<std::exception, TToken>::value>::type * = nullptr>
+    basic_formatter &operator<<(const TToken &s) {
+        underlying << s.what();
         return *this;
     }
 
@@ -135,13 +147,13 @@ public:
     // comma operator overloaded as well, choose your preferred way.
     template <typename TToken>
     const basic_formatter& operator, (const TToken& s) const {
-        underlying << s;
+        *this << s;
         return *this;
     }
 
     template <typename TToken>
     basic_formatter& operator, (const TToken& s) {
-        underlying << s;
+        *this << s;
         return *this;
     }
 
@@ -149,7 +161,7 @@ public:
     // See https://sourceforge.net/projects/assimp/forums/forum/817654/topic/4372824
     template <typename TToken>
     basic_formatter& operator, (TToken& s) {
-        underlying << s;
+        *this << s;
         return *this;
     }
 

